util/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/monitoring/MonitoringService.java

```diff
@@ -0,0 +1,175 @@
+    }
+    this.monitoredEvents = monitoredEvents;
+
+    // configure Resource
+    MonitoredResource.Builder resourceBuilder = MonitoredResource.newBuilder();
+    Map<String, String> resourceLabels = new HashMap<>();
+    resourceLabels.put("project_id", this.projectId);
+
+    Map<String, String> env = System.getenv();
+    if (env.containsKey("ENV_POD_NAME")
+        && env.containsKey("ENV_POD_NAMESPACE")
+        && env.containsKey("ENV_CONTAINER_NAME")) {
+      resourceLabels.put("pod_name", env.get("ENV_POD_NAME"));
+      resourceLabels.put("namespace_name", env.get("ENV_POD_NAMESPACE"));
+      resourceLabels.put("container_name", env.get("ENV_CONTAINER_NAME"));
+      resourceLabels.put("cluster_name", GcpMetadataUtil.get(requestFactory, META_CLUSTER_NAME));
+      String location = GcpMetadataUtil.get(requestFactory, META_LOCATION);
+      location = location.substring(location.lastIndexOf('/') + 1);
+      resourceLabels.put("location", location);
+      resourceBuilder.setType("k8s_container");
+    } else {
+      resourceBuilder.setType("global");
+    }
+
+    this.monitoredResource = resourceBuilder.putAllLabels(resourceLabels).build();
+    log.info("monitoredResource = {}", monitoredResource);
+  }
+
+  public static void initialize(String projectId, IMonitoringEvent[] monitoredEvents,
+      HttpRequestFactory requestFactory) {
+    if (INSTANCE != null) {
+      throw new IllegalStateException("Already initialized");
+    }
+    INSTANCE = new MonitoringService(projectId, monitoredEvents, requestFactory);
+  }
+
+  public static void addEvent(IMonitoringEvent eventType, long value) {
+    if (INSTANCE == null) {
+      log.warn("MonitoringService not initialized, skipping: {}={}", eventType, value);
+    } else {
+      INSTANCE._addEvent(eventType, value);
+    }
+  }
+
+  public static void addEvent(IMonitoringEvent eventType) {
+    addEvent(eventType, 1L);
+  }
+
+  private void _addEvent(IMonitoringEvent eventType, long value) {
+    synchronized (aggregateEvents) {
+      long prevValue = aggregateEvents.getOrDefault(eventType, 0L);
+      aggregateEvents.put(eventType, prevValue + value);
+    }
+  }
+
+  private void flush() {
+    HashMap<IMonitoringEvent, Long> flushEvents = null;
+    synchronized (aggregateEvents) {
+      flushEvents = new HashMap<>(aggregateEvents);
+      aggregateEvents.clear();
+    }
+
+    try {
+      Timestamp flushTime = Timestamps.fromMillis(System.currentTimeMillis());
+
+      List<TimeSeries> timeSeriesList = new ArrayList<>();
+      for (IMonitoringEvent event : monitoredEvents) {
+        TimeInterval interval = TimeInterval.newBuilder()
+            .setEndTime(flushTime)
+            .build();
+        TypedValue value = TypedValue.newBuilder()
+            .setInt64Value(flushEvents.getOrDefault(event, 0L))
+            .build();
+        Point point = Point.newBuilder()
+            .setInterval(interval)
+            .setValue(value)
+            .build();
```

Comment:
Yes, I tried not sending empty data points initially, but found no way to build usable graphs from them (would need to be built with assumption that no data point for period = 0, instead of only connecting submitted data points).

Reply:
Graph when submitting 0 time series:
![metric_explorer_graph](https://user-images.githubusercontent.com/46771951/60420954-08f38700-9c13-11e9-84ed-d669ef05439f.png)
Graph when submitting only actual data points (right part, left is densely populated data from previous image):
![metric_explorer_sparse](https://user-images.githubusercontent.com/46771951/60421018-29234600-9c13-11e9-91e3-1c984515af30.png)
Same, but visualized as stacked bars:
![metric_explorer_sparse__stacked_bar](https://user-images.githubusercontent.com/46771951/60421070-4526e780-9c13-11e9-851f-3818a93bcf7c.png)